File: ws2/code/test/FilterCoeff.c
#define SAMPLE_RATE 24000.0
#define PI 3.14159265359

static float BW = 0.1 * PI;     // Bandwidth
static float F = 550;           // Central frequency

static double alpha, beta;      // filter coefficients
static double a1, a2;
static double b0, b1, b2;

void FilterCoeff(void) {
    beta = cos(F * 2 * PI / SAMPLE_RATE);

    double cosine = cos(BW);
    alpha = 1/cosine - sqrt(1/(cosine*cosine) - 1);

    printf("alpha=%f\nbeta=%f\n", alpha, beta);

    double coefficient = (1 + alpha) / 2;
    a1 = -beta * (1 + alpha);
    a2 = alpha;
    b0 = coefficient;
    b1 = -2 * beta * coefficient;
    b2 = coefficient;
}

File: ws2/code/test/FilterCoeff.m
clear;
close all;

omega0 = 550 * 2 * pi / 24E3;
beta = cos(omega0);

fprintf('beta=%f\n', beta);


for bw = [0.1 0.01 0.0025]
    cosine = cos(bw * pi);
    alpha = 0.5 * (2/cosine - sqrt(4/cosine^2 -4));

    fprintf('alpha=%f (Bw = %.4fpi)\n', alpha, bw);

    num = ((1+alpha) / 2) * [1 -2*beta 1];
    den = [1 -beta*(1+alpha) alpha];
end
